packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -143,28 +1362,56 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
    */
   readonly routeType: Protocol;
 
+  /**
+   * The gateway route action to rewrite.
+   *
+   * @default - rewritten to the target Virtual Service's hostname and the matched prefix or path is rewritten to '/'.
+   */
+  readonly rewrite?: HttpGatewayRouteRewrite;
+
   constructor(options: HttpGatewayRouteSpecOptions, protocol: Protocol.HTTP | Protocol.HTTP2) {
     super();
     this.routeTarget = options.routeTarget;
     this.routeType = protocol;
     this.match = options.match;
+    this.rewrite = options.rewrite;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    const prefixPath = this.match ? this.match.bind(scope).requestMatch.prefix : '/';
+
+    if (prefixPath && prefixPath[0] != '/') {
       throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
     }
+    if (this.rewrite?.bind(scope).prefix && !this.match?.bind(scope).requestMatch.prefix) {
+      throw new Error('HTTP Gateway Route Prefix Match must be set.');
+    }
+    if (this.rewrite?.bind(scope).path && this.match?.bind(scope).requestMatch.prefix) {
+      throw new Error('HTTP Gateway Route Prefix Match and Path Rewrite both cannot be set.');
+    }
+
     const httpConfig: CfnGatewayRoute.HttpGatewayRouteProperty = {
       match: {
         prefix: prefixPath,
+        headers: this.match?.bind(scope).requestMatch.headers,
+        hostname: this.match?.bind(scope).requestMatch.hostname,
+        method: this.match?.bind(scope).requestMatch.method,
+        path: this.match?.bind(scope).requestMatch.path,
+        queryParameters: this.match?.bind(scope).requestMatch.queryParameters,
```

Comment:
`match` property is not longer `optional`. It is `required`.

Reply:
We can make this optional still and just default to `.prefix('/')` imo

Reply:
That makes sense and I like less breaking change :) Changing to something like

```typescript
match: this.match
        ? {
          ...this.match.bind(scope).requestMatch,
          prefix: prefixPath,
        }: {
          prefix: prefixPath,
        },
```